<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  
  
  <link rel="shortcut icon" href="../img/favicon.ico">
  <title>HRD Score - Documentation</title>
  <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Lato:400,700|Roboto+Slab:400,700|Inconsolata:400,700" />

  <link rel="stylesheet" href="../css/theme.css" />
  <link rel="stylesheet" href="../css/theme_extra.css" />
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/github.min.css" />
  
  <script>
    // Current page data
    var mkdocs_page_name = "HRD Score";
    var mkdocs_page_input_path = "HRD_score.md";
    var mkdocs_page_url = null;
  </script>
  
  <script src="../js/jquery-2.1.1.min.js" defer></script>
  <script src="../js/modernizr-2.8.3.min.js" defer></script>
  <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js"></script>
  <script>hljs.initHighlightingOnLoad();</script> 
  
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
    <div class="wy-side-scroll">
      <div class="wy-side-nav-search">
        <a href=".." class="icon icon-home"> Documentation</a>
        <div role="search">
  <form id ="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" title="Type search term here" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="..">Home</a>
                    </li>
                </ul>
                <ul class="current">
                    <li class="toctree-l1 current"><a class="reference internal current" href="./">HRD Score</a>
    <ul class="current">
    <li class="toctree-l2"><a class="reference internal" href="#workflow-overview">Workflow overview</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#sequenza">Sequenza</a>
        <ul>
    <li class="toctree-l3"><a class="reference internal" href="#preprocessing-of-input-files">Preprocessing of input files</a>
    </li>
        </ul>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#scarhrd">scarHRD</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#genomic-scar-scores">Genomic scar scores</a>
        <ul>
    <li class="toctree-l3"><a class="reference internal" href="#loss-of-heterozygosity-hrd-loh">Loss of Heterozygosity (HRD-LOH)</a>
    </li>
    <li class="toctree-l3"><a class="reference internal" href="#number-of-telomeric-allelic-imbalances">Number of Telomeric Allelic Imbalances</a>
    </li>
        </ul>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#references">References</a>
    </li>
    </ul>
                    </li>
                </ul>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../git/">Git</a>
                    </li>
                </ul>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../Yaml_files/">Yaml</a>
                    </li>
                </ul>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../remark_boilerplate.html">Markdown</a>
                    </li>
                </ul>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../Conda/">Conda</a>
                    </li>
                </ul>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../Django/">Django</a>
                    </li>
                </ul>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../why_django.html">Why Django?</a>
                    </li>
                </ul>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../Docker/">Docker</a>
                    </li>
                </ul>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../Miracum_pipe_galaxy/">Miracum Galaxy</a>
                    </li>
                </ul>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../MAF/">MAF</a>
                    </li>
                </ul>
      </div>
    </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="..">Documentation</a>
      </nav>

      
      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="..">Docs</a> &raquo;</li>
    
      
    
    <li>HRD Score</li>
    <li class="wy-breadcrumbs-aside">
      
    </li>
  </ul>
  
  <hr/>
</div>
          <div role="main">
            <div class="section">
              
                <h1 id="hrd-score">HRD Score</h1>
<p>Task: Determine HRD Score</p>
<p>Solution: scarHRD package (<a href="https://github.com/sztup/scarHRD#running-on-mouse-genomes">https://github.com/sztup/scarHRD#running-on-mouse-genomes</a>)</p>
<p>Description:</p>
<p><code>scarHRD</code> is an R package which determines the levels of homologous recombination deficiency (telomeric allelic imbalance, loss off heterozygosity, number of large-scale transitions) based on NGS (WES, WGS) data.</p>
<p>The first genomic scar based homologous recombination deficiency measures were produced using SNP arrays. Since this technology has been largely replaced by next generation sequencing it has become important to develop algorithms that derive the same type of genomic scar-scores from next generation sequencing (WXS, WGS) data. In order to perform this analysis, here we introduce the scarHRD R package and show that using this method the SNP-array based and next generation sequencing based derivation of HRD scores show good correlation.</p>
<h2 id="workflow-overview">Workflow overview</h2>
<p>A typical workflow of determining the genomic scar scores for a tumor sample has the following steps:</p>
<ol>
<li>
<p>Call allele specific copy number profile on paired normal-tumor BAM files. This step has to be executed before running scarHRD. We recommend using Sequenza (Favero et al. 2015) <a href="http://www.cbs.dtu.dk/biotools/sequenza/">http://www.cbs.dtu.dk/biotools/sequenza/</a> for copy number segmentation, Other tools (e.g. ASCAT (Van Loo et al. 2010)) may also be used in this step.
   This step is time-consuming and compute-intensive.</p>
</li>
<li>
<p>Determine the scar scores with scarHRD R package.
   This step only takes a few minutes.</p>
</li>
</ol>
<h2 id="sequenza">Sequenza</h2>
<h3 id="preprocessing-of-input-files">Preprocessing of input files</h3>
<p>In order to obtain precise mutational and aberration patterns in a tumor sample, Sequenza requires a matched normal sample from the same patient. Typically, the following files are needed to get started with Sequenza:</p>
<ul>
<li>A BAM file (or a derived pileup file) from the tumor specimen.</li>
<li>A BAM file (or a derived pileup file) from the normal specimen.</li>
<li>A FASTA reference genomic sequence file</li>
</ul>
<p>The normal and tumor BAM files are processed together to generate a seqz file, which is the required input for the analysis. It is possible to generate a seqz starting from other processed data, such as pileup, or VCF files. The available options are described in the sequenza-utils manual pages.</p>
<p>The sequenza-utils command provides various tools; here we highlight only the basic usage:</p>
<p>Process a FASTA file to produce a GC Wiggle track file (has to be done only once):</p>
<pre><code class="language-console">sequenza−utils gc_wiggle −w 50 --fasta hg19.fa -o hg19.gc50Base.wig.gz
</code></pre>
<p>Process BAM and Wiggle files to produce a seqz file:</p>
<pre><code class="language-console">sequenza−utils bam2seqz -n normal.bam -t tumor.bam --fasta hg19.fa \
    -gc hg19.gc50Base.wig.gz -o out.seqz.gz
</code></pre>
<p>Post-process by binning the original seqz file:</p>
<pre><code class="language-console">sequenza−utils seqz_binning --seqz out.seqz.gz -w 50 -o out small.seqz.gz
</code></pre>
<p>The <code>small.seqz.gz</code> would be the input to <code>scarHRD</code>.</p>
<h2 id="scarhrd">scarHRD</h2>
<p>The scarHRD input may be a detailed segmentation file from Sequenza, in case there is a reliable estimation of ploidy of the tumor sample is known, it should be sumbitted in the ploidy argument of the scarHRD function, otherwise ploidy between 1 and 5.5 will be tested:</p>
<pre><code class="language-R">a&lt;-read.table(&quot;/examples/test1.small.seqz.gz&quot;, header=T)
head(a)
</code></pre>
<pre><code class="language-R">##   chromosome position base.ref depth.normal depth.tumor depth.ratio    Af
## 1       chr1    12975        N            7          20       2.841 1.000
## 2       chr1    13020        A            8          28       3.500 0.964
## 3       chr1    13026        N           15          43       2.964 1.000
## 4       chr1    13038        T           11          35       3.182 0.971
## 5       chr1    13041        A           11          37       3.364 0.946
## 6       chr1    13077        N           26          65       2.465 1.000
##   Bf zygosity.normal GC.percent good.reads AB.normal AB.tumor tumor.strand
## 1  0             hom         60         51         N        .            0
## 2  0             hom         60         28         A   G0.036         G1.0
## 3  0             hom         59         51         N        .            0
## 4  0             hom         59         35         T   C0.029         C1.0
## 5  0             hom         59         37         A   G0.054         G0.5
## 6  0             hom         62         51         N        .            0
</code></pre>
<p>Usage example:</p>
<pre><code class="language-R">library(&quot;scarHRD&quot;)
scar_score(&quot;F:/Documents/scarHRD/examples/test1.small.seqz.gz&quot;,reference = &quot;grch38&quot;, seqz=TRUE)
</code></pre>
<pre><code class="language-R">## Preprocessing started...

## Processing chr1: 18 variant calls; 6290 heterozygous positions; 549112 homozygous positions.
## Processing chr2: 22 variant calls; 4934 heterozygous positions; 394216 homozygous positions.

##
  |=================================================================| 100%
## Preprocessing finished
## Determining HRD-LOH, LST, TAI

##      HRD Telomeric AI LST HRD-sum
## [1,]   1            2   0       3

</code></pre>
<pre><code class="language-R">scar_score(&quot;F:/Documents/scarHRD/examples/test2.txt&quot;,reference = &quot;grch38&quot;, seqz=FALSE)
</code></pre>
<pre><code class="language-R">## Determining HRD-LOH, LST, TAI

##      HRD Telomeric AI LST HRD-sum
## [1,]  25           35  33      93
</code></pre>
<p>Parameters</p>
<p><code>seg</code> -- input file name
<code>reference</code> -- the reference genome used, <code>grch38</code> or <code>grch37</code> or mouse (default: <code>grch38</code>)
<code>seqz</code> -- <code>TRUE</code> if the input file is a smallo.seqz.gz file, otherwise <code>FALSE</code> (default: <code>TRUE</code>)
<code>ploidy</code> -- optional, previously estimated ploidy of the sample <code>outputdir</code> -- optional, the path to the output directory
<code>chr.in.names</code> -- optional, default: TRUE, set to FALSE if input file does not contain 'chr' in chromosome names.</p>
<h2 id="genomic-scar-scores">Genomic scar scores</h2>
<h3 id="loss-of-heterozygosity-hrd-loh">Loss of Heterozygosity (HRD-LOH)</h3>
<p>The HRD-LOH score was described based on investigation in SNP-array-based copy number profiles of ovarian cancer (Abkevich et al. 2012). In this paper the authors showed that the samples with deficient BRCA1, BRCA2 have higher HRD-LOH scores compared to BRCA-intact samples, thus this measurement may be a reliable tool to estimate the sample's homologous recombination capacity.
The definition of a sample's HRD-LOH score is the number of 15 Mb exceeding LOH regions which do not cover the whole chromosome. In the first paper publishing HRD-LOH-score (Abkevich et al., 2012) the authors examine the correlation between HRD-LOH-score and HR deficiency calculated for different LOH region length cut-offs. In that paper the cut-off of 15 Mb approximately in the middle of the interval was arbitrarily selected for further analysis. The authors argue that the rational for this selection rather than selecting the cut-off with the lowest p-value is that the latter cut-off is more sensitive to statistical noise present in the data.
In our manuscript we also investigated if this 15 Mb cutoff is appropriate for WXS-based HRD-LOH score.We followed the same principles as Abkievits et al, thus while there was small difference between the p-values for the different minimum length cutoff values, we chose to use the same, 15 Mb limit as Abkevich et al. We also performed Spearman rank correlation between the SNP-array-based and WXS-based HRD-LOH scores for the different cutoff minimum LOH length cutoff (manuscript, Supplementary Figure S3C). Here the 14 Mb and 15 Mb cutoff-based WXS-HRD-LOH score had the highest correlation with the SNP-based HRD score. (0.700 and 0.695 respectively). This result reassured our choice of using the 15 Mb cutoff like in the SNP-array-based HRD-LOH score.</p>
<p><img alt="alt text" src="https://raw.githubusercontent.com/sztup/scarHRD/master/vignettes/hrd-loh.svg" title="hrd-loh" /></p>
<p>Figure 1.A Visual representation of the HRD-LOH score on short theoretical chromosomes. Figure 1.B: Calculating HRD-LOH from a biallelic copy-number profile; LOH regions a, and c, would both increase the score by 1, while neither b, or d, would add to its value (b, does not pass the length requirement, and d covers a whole chromosome)</p>
<p>Large Scale Transitions (LST)
The presence of Large Scale Transitions in connection with homologous recombination deficiency was first studied in basal-like breast cancer (Popova et al. 2012). Based on SNP-array derived copy number profiles BRCA1-inactivated cases had showed higher number of large scale transitions.
A large scale transition is defined as a chromosomal break between adjacent regions of at least 10 Mb, with a distance between them not larger than 3Mb.</p>
<p><img alt="alt text" src="https://raw.githubusercontent.com/sztup/scarHRD/master/vignettes/lst.svg" title="lst" /></p>
<p>Figure 2.A: Visual representation of the LST score on short theoretical chromosomes. Figure 2.B: Calculating LST scores from a biallelic copy-number profile; events that are marked with green "marked" signs would increase the score, while events marked with red crosses would not. The grey areas represent the centromeric regions. (From left to right; Chromosome 1: the first event passes the definition of an LST, the second bounded by a shorter than 10 Mb segment from the right, the third is bounded by a segment from the left, which extends to the centromere, the fourth’s gap is greater than 3 Mb. Chromosome 2: The first event is a valid LST, the second and third are not because they are bounded by centromeric segments, and the fourth is a valid LST)</p>
<h3 id="number-of-telomeric-allelic-imbalances">Number of Telomeric Allelic Imbalances</h3>
<p>Allelic imbalance (AI) is the unequal contribution of parental allele sequences with or without changes in the overall copy number of the region. Our group have previously found, that the number telomeric AIs is indicative of defective DNA repair in ovarian cancer and triple-negative breast cancer, and that higher number of telomeric AI is associated with better response to cisplatin treatment (Birkbak et al. 2012).
The number of telomeric allelic imbalances is the number AIs that extend to the telomeric end of a chromosome.</p>
<p><img alt="alt text" src="https://raw.githubusercontent.com/sztup/scarHRD/master/vignettes/ntai.svg" title="ntai" /></p>
<p>Figure 3.A: Visual representation of the ntAI on short theoretical chromosomes. Figure 3.B: Illustration of possible telomeric allelic imbalances in an allele specific copy number profile.</p>
<h2 id="references">References</h2>
<p><a href="https://github.com/sztup/scarHRD#running-on-mouse-genomes">https://github.com/sztup/scarHRD#running-on-mouse-genomes</a></p>
              
            </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="../git/" class="btn btn-neutral float-right" title="Git">Next <span class="icon icon-circle-arrow-right"></span></a>
      
      
        <a href=".." class="btn btn-neutral" title="Home"><span class="icon icon-circle-arrow-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
    
  </div>

  Built with <a href="https://www.mkdocs.org/">MkDocs</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
      
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" aria-label="versions">
    <span class="rst-current-version" data-toggle="rst-current-version">
      
      
        <span><a href=".." style="color: #fcfcfc;">&laquo; Previous</a></span>
      
      
        <span style="margin-left: 15px"><a href="../git/" style="color: #fcfcfc">Next &raquo;</a></span>
      
    </span>
</div>
    <script>var base_url = '..';</script>
    <script src="../js/theme.js" defer></script>
      <script src="../search/main.js" defer></script>
    <script defer>
        window.onload = function () {
            SphinxRtdTheme.Navigation.enable(true);
        };
    </script>

</body>
</html>
